packages/calcite-components/src/components/tooltip/tooltip.e2e.ts

```diff
@@ -1039,36 +1066,65 @@ describe("calcite-tooltip", () => {
           );
         }
 
-        document.body.innerHTML = "<shadow-component-b></shadow-component-b>";
+        document.body.innerHTML = `<shadow-component-b id="one"></shadow-component-b>
+        <shadow-component-b id="two"></shadow-component-b>`;
       });
       await page.waitForChanges();
     }
 
-    function isTooltipOpen(page: E2EPage): Promise<boolean> {
-      return page.evaluate((): boolean => {
+    function isTooltipOpen(page: E2EPage, componentId = "one"): Promise<boolean> {
+      return page.evaluate((componentId): boolean => {
         return document
-          .querySelector("shadow-component-b")
+          .querySelector(`#${componentId}`)
           .shadowRoot.querySelector("shadow-component-a")
           .shadowRoot.querySelector("calcite-tooltip").open;
-      });
+      }, componentId);
     }
 
-    async function focusReferenceElement(page: E2EPage): Promise<void> {
-      await page.evaluate((): void => {
-        const referenceElement = document
-          .querySelector("shadow-component-b")
-          .shadowRoot.querySelector("shadow-component-a")
-          .shadowRoot.querySelector("button");
-        referenceElement.dispatchEvent(new FocusEvent("focusin"));
-      });
+    async function focusReferenceElement(page: E2EPage, componentId = "one"): Promise<void> {
+      await page.evaluate(
```

Comment:
I _think_ you could simplify this by using Stencil's updated deep piercing feature:

```tsx
// ⚠️ untested ⚠️
await page.$eval(`#${componentId} >>> shadow-component-a >>> button`, (refEl, eventOptions) => {
  refEl.dispatchEvent(new FocusEvent("focusin", eventOptions));
}, eventOptions);
```